quint/src/runtime/impl/compilerImpl.ts

```diff
@@ -252,10 +255,16 @@ export class CompilerVisitor implements IRVisitor {
       // Both input1 and input2 wrap step, but in their individual computables.
       const unwrappedValue = boundValue
       const app: ir.QuintApp = { id: opdef.id, kind: 'app', opcode: opdef.name, args: [] }
+      const evalApp: ir.QuintApp = { id: 0n, kind: 'app', opcode: '_', args: [app] }
       boundValue = {
         eval: () => {
-          this.execListener.onUserOperatorCall(app)
-          const r = unwrappedValue.eval()
+          if (app.opcode === inputDefName) {
+            this.execListener.onUserOperatorCall(evalApp)
+            // do not return from '_', as it may span over multiple frames
+          } else {
+            this.execListener.onUserOperatorCall(app)
+          }
```

Comment:
I'm not sure how to read this comment. It seems like if `if` branch "does not return" then neither would the `else` branch?

Reply:
Good point. I will fix that.